docs: update component content

josmar-crwdstffng opened the pull request:
Reorganize component content to be more module and align with 1 content type per topic.

Moved all component-related content under the `guide/component` directory.

Updated introductory content in component-overview topic. 
Created new implementation tasks in component-create topic. 
Created new usage tasks in component-usage topic. 
Moved tutorial content in new component-example topic. 
Moved and reorganized content in new task topics.
Updated links. 
Cleaned up markdown component topics.

## PR Checklist

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

The changes do not affect implementation of code, but do affect the organization of content

josmar-crwdstffng commented:
@TMDavisGoogle 
Please perform an edit pass at your earliest convenience.
Thank you

mary-poppins commented:
You can preview 06ea065 at https://pr45501-06ea065.ngbuilds.io/.

TMDavisGoogle commented on aio/content/errors/NG0100.md:
```diff
@@ -1,20 +1,35 @@
+Angular throws the error to ensure data is always reflected correctly in the view, which prevents erratic UI behavior or a possible infinite loop.
 
-This error commonly occurs when you’ve added template expressions or begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`. It is also common when dealing with loading status and asynchronous operations, or a child component changes its parent bindings.
+This error commonly occurs when you have added template expressions or begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`.
```
```suggestion
This error commonly occurs when you have added template expressions or have begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`.
```

josmar-crwdstffng replied:
Updated.

```markdown
This error commonly occurs when you have added template expressions or have begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`.
```

dgp1130 commented:
@jelbourn should we use a global approval for this? It touches a lot of things but doesn't seem to be changing much of the semantics.

josmar-crwdstffng commented on aio/content/errors/NG0100.md:
```diff
@@ -1,20 +1,35 @@
 
-This error commonly occurs when you’ve added template expressions or begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`. It is also common when dealing with loading status and asynchronous operations, or a child component changes its parent bindings.
+This error commonly occurs when you have added template expressions or begun to implement lifecycle hooks like `ngAfterViewInit` or `ngOnChanges`.
+It is also common when dealing with loading status and asynchronous operations, or a child component changes its parent bindings.
```
Updated.

```markdown
It is also common when dealing with loading status and asynchronous operations, or when a child component changes its parent bindings.
```

mary-poppins commented:
You can preview ff289e8 at https://pr45501-ff289e8.ngbuilds.io/.

mary-poppins commented:
You can preview 5849e27 at https://pr45501-5849e27.ngbuilds.io/.

TMDavisGoogle reviewed (changes requested):
Punctuation, grammar, spelling edits for clarity.

josmar-crwdstffng commented on aio/content/guide/component-overview-old.md:
`````diff
@@ -0,0 +1,189 @@
+## old content
+
+````markdown
+Components are the main building block for Angular applications. Each component consists of:
`````
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+&num;&num; Special selectors
+
+Component styles have a few special &ast;selectors&ast; from the world of shadow DOM style scoping
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+Use the &ast;function form&ast; to apply host styles conditionally by
+including another selector inside parentheses after &grave;:host&grave;.
+
+In this example the host's content also becomes bold when the &grave;active&grave; CSS class is applied to the host element.
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+&lt;code-example path="component-styles/src/app/hero-details.component.css" region="hostfunction" header="src/app/hero-details.component.css"&gt;&lt;/code-example&gt;
+
+The &grave;:host&grave; selector can also be combined with other selectors.
+Add selectors behind the &grave;:host&grave; to select child elements, for example using &grave;:host h2&grave; to target all &grave;&lt;h2&gt;&grave; elements inside a component's view.
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+Sometimes it's useful to apply styles to elements within a component's template 
+based on some condition in an element that is an ancestor of the host element.
+For example, a CSS theme class could be applied to the document &grave;&lt;body&gt;&grave; element, and
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+Component styles normally apply only to the HTML in the component's own template.
+
+Applying the &grave;::ng-deep&grave; pseudo-class to any CSS rule completely disables view-encapsulation for
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+&lt;div class="alert is-important"&gt;
+
+The shadow-piercing descendant combinator is deprecated and \[support is being removed from major browsers\]\(https://www.chromestatus.com/feature/6750456638341120\) and tools.
+As such we plan to drop support in Angular \(for all 3 of &grave;/deep/&grave;, &grave;&gt;&gt;&gt;&grave; and &grave;::ng-deep&grave;\).
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+&lt;div class="alert is-critical"&gt;
+
+When building with the CLI, be sure to include the linked style file among the assets to be copied to the server as described in the \[Assets configuration guide\]\(guide/workspace-config&num;assets-configuration\).
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+When building with the CLI, be sure to include the linked style file among the assets to be copied to the server as described in the \[Assets configuration guide\]\(guide/workspace-config&num;assets-configuration\).
+
+Once included, the CLI includes the stylesheet, whether the link tag's href URL is relative to the application root or the component file.
```
This topic is being removed.

josmar-crwdstffng commented on aio/content/guide/component/component-dynamic-load.md:
```diff
@@ -0,0 +1,175 @@
+
+</code-example>
+
+Use the `loadComponent()` method complete the majority of the work for you.
```
Updated.

```markdown
Use the `loadComponent()` method to complete the majority of the work for you.
```

josmar-crwdstffng commented on aio/content/guide/component/component-usage-complex-content-projection.md:
```diff
@@ -0,0 +1,64 @@
+# Project content in more complex environments
+
+Typically usage for [Multi-slot Content Projection][AioGuideComponentUsageMultiSlotContentProjection] include using an attribute, element, style defintion, or some combination of all three to identify where to project your content.
```
Updated.

```markdown
Typically usage for [Multi-slot Content Projection][AioGuideComponentUsageMultiSlotContentProjection] includes using an attribute, element, style definition, or some combination of all three to identify where to project your content.
```

mary-poppins commented:
You can preview 02f32be at https://pr45501-02f32be.ngbuilds.io/.

josmar-crwdstffng commented:
@TMDavisGoogle 
All edit review comments were addressed or applied.
Please verify.
Thank you

mary-poppins commented:
You can preview e9d83b3 at https://pr45501-e9d83b3.ngbuilds.io/.

jessicajaniuk commented:
Looks like since the most recent docs PR merge, this needs a rebase.

mary-poppins commented:
You can preview bf32e0c at https://pr45501-bf32e0c.ngbuilds.io/.

TMDavisGoogle commented on aio/content/guide/component/component-lifecycle.md:
```diff
@@ -0,0 +1,35 @@
+# Understand the lifecycle of a component
+
+The lifecycle of a Angular component starts when the Angular instantiates the component class and ends when Angular destroys the instance of the component.
```
```suggestion
The lifecycle of a Angular component starts when the Angular instantiates the component class, and ends when Angular destroys the instance of the component.
```

TMDavisGoogle commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+# Angular component overview
+
+A [component][AioGuideGlossaryComponent] is the main building block for an Angular application.
+Every Angular application has at least one component, the *root component* that connects a component hierarchy with the document object model \(DOM\) of the page.
```
```suggestion
Every Angular application has at least one component, the *root component*, that connects a component hierarchy with the document object model \(DOM\) of the page.
```

josmar-crwdstffng replied:
Updated.

```markdown
Every Angular application has at least one component, the *root component*, that connects a component hierarchy with the document object model \(DOM\) of the page.
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-conditional-content-projection.md:
```diff
@@ -0,0 +1,152 @@
+
+If your component needs to conditionally render content, or render content multiple times, you should configure that component to accept an `ng-template` element that contains the content you want to conditionally render.
+
+Using an `ng-content` element in these cases is not recommended, because when the consumer of a component supplies the content, that content is always initialized, even if the component does not define an `ng-content` element or if that `ng-content` element is inside of an `ngIf` statement.
```
```suggestion
Using an `ng-content` element in these cases is not recommended because when the consumer of a component supplies the content, that content is always initialized, even if the component does not define an `ng-content` element or if that `ng-content` element is inside of an `ngIf` statement.
```

josmar-crwdstffng replied:
Updated.

```markdown
Use of an `ng-content` element in these cases is not recommended, because when the consumer of a component supplies the content.
That content supplied by the consumer is always initialized, even if the component does not define an `ng-content` element or if that `ng-content` element is inside of an `ngIf` statement.
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-exchange-data.md:
```diff
@@ -0,0 +1,57 @@
+
+## Prerequisites
+
+Before you exchange data between to a child component and a parent component in an Angular [component][AioGuideGlossaryComponent], verify that you have met the following prerequisites.
```
```suggestion
Before you exchange data between a child component and a parent component in an Angular [component][AioGuideGlossaryComponent], verify that you have met the following prerequisites.
```

josmar-crwdstffng replied:
Updated.

```markdown
Before you exchange data between a child component and a parent component in an Angular [component][AioGuideGlossaryComponent], verify that you have met the following prerequisites.
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-lifecycle-hooks.md:
```diff
@@ -0,0 +1,142 @@
+# Use an Angular lifecycle hook method
+
+Use a [lifecycle hook method][AioGuideGlossaryLifecycleHook] to tap into key events in the lifecycle of a component or directive and complete the following actions.
```
```suggestion
Use a [lifecycle hook method][AioGuideGlossaryLifecycleHook] to tap into key events in the lifecycle of a component or directive, and complete the following actions.
```

josmar-crwdstffng replied:
Updated.

```markdown
Use a [lifecycle hook method][AioGuideGlossaryLifecycleHook] to tap into key events in the lifecycle of a component or directive, and complete the following actions.
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-lifecycle-hooks.md:
```diff
@@ -0,0 +1,142 @@
+
+| Task                                                         | Details |
+|:---                                                          |:---     |
+| Perform complex initializations outside of the constructor   | Components should be cheap and safe to construct. Do not fetch data in a component constructor. Do not worry that a new component tries to contact a remote server when created under test or before you decide to display it. <br /> An `ngOnInit()` hook method is a good place for a component to fetch the initial data. For an example, s ...
```
```suggestion
| Perform complex initializations outside of the constructor   | Components should be cheap and safe to construct. Do not fetch data in a component constructor. Do not worry that a new component tries to contact a remote server when created under test, or before you decide to display it. <br /> An `ngOnInit()` hook method is a good place for a component to fetch the initial data. For an example, see the [Tour of Heroes tutorial][AioTutorialTohPt4CallItInNgoninit].                                                                                                                                                                  |
```

josmar-crwdstffng replied:
Updated.

```markdown
| Perform complex initializations outside of the constructor   | Components should be cheap and safe to construct. Do not fetch data in a component constructor. Do not worry that a new component tries to contact a remote server when created under test, or before you decide to display it. <br /> An `ngOnInit()` hook method is a good place for a component to fetch the initial data. For an example, see the [Tour of Heroes tutorial][AioTutorialTohPt4CallItInNgoninit].                                                                                                                                                                  |
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-multi-slot-content-projection.md:
```diff
@@ -0,0 +1,96 @@
+# Multi-slot content projection
+
+A component is able to use multiple slots.
+Each slot is able to specify a CSS selector that determines the content goes into each slot.
```
```suggestion
Each slot is able to specify a CSS selector that determines the content that goes into each slot.
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-send-data-to-parent.md:
```diff
@@ -0,0 +1,202 @@
+    | Parts                           | Details |
+    |:---                             |:---     |
+    | `@Output()`                     | A decorator function. Marks the property as a way for data to go from the child to the parent.                                                                     |
+    | `new{nameOfChildProperty}Event` | The name of the property. The `new` and `Event` terms were used to indicate the property emits a new event.                                                        |
```
```suggestion
    | `new{nameOfChildProperty}Event` | The name of the property. The `new` and `Event` terms were used to indicate that the property emits a new event.                                                        |
```

josmar-crwdstffng replied:
Updated.

```markdown
| `new{nameOfChildProperty}Event` | The name of the property. The `new` and `Event` terms were used to indicate that the property emits a new event.                                                   |
```

TMDavisGoogle commented on aio/content/guide/component/component-usage-send-data-to-parent.md:
```diff
@@ -0,0 +1,202 @@
+    </code-example>
+
+    The `addNew{nameOfChildProperty}()` method uses the `new{nameOfChildProperty}Event` `@Output()` decorator function to raise an event.
+    It uses the value the user types into the `input` element.
```
```suggestion
    It uses the value that the user types into the `input` element.
```

josmar-crwdstffng replied:
Updated.

```markdown
It uses the value that the user types into the `input` element.
```

TMDavisGoogle commented on aio/content/guide/dynamic-component-loader.md:
```diff
@@ -1,130 +1,151 @@
-The `loadComponent()` method is doing a lot of the heavy lifting here.
-Take it step by step.
-First, it picks an ad.
+The &grave;loadComponent\(\)&grave; method is doing a lot of the heavy lifting here.
```
```suggestion
The &grave;loadComponent\(\)&grave; method is doing a lot of the work here.
```

TMDavisGoogle commented on aio/content/guide/lightweight-injection-tokens.md:
```diff
@@ -134,30 +154,31 @@ class LibCardComponent {
-This lets full tree shaking of `LibHeaderComponent` take place.
-The `LibHeaderToken` is retained, but it is only a class declaration, with no concrete implementation.
+In this example, the `LibCardComponent` component implementation no longer refers to the `LibHeaderComponent` component in either the type position or the value position.
+This lets full tree shaking of `LibHeaderComponent` component take place.
```
```suggestion
This lets full tree-shaking of `LibHeaderComponent` component take place.
```

josmar-crwdstffng replied:
Updated.

```markdown
This allows full tree-shaking of `LibHeaderComponent` component take place.
```

TMDavisGoogle commented on aio/content/guide/lightweight-injection-tokens.md:
```diff
@@ -134,30 +154,31 @@ class LibCardComponent {
-This lets full tree shaking of `LibHeaderComponent` take place.
-The `LibHeaderToken` is retained, but it is only a class declaration, with no concrete implementation.
+In this example, the `LibCardComponent` component implementation no longer refers to the `LibHeaderComponent` component in either the type position or the value position.
+This lets full tree shaking of `LibHeaderComponent` component take place.
```
```suggestion
This lets full tree-shaking of `LibHeaderComponent` component take place.
```

TMDavisGoogle commented on aio/content/guide/testing-components-basics.md:
```diff
@@ -25,7 +25,7 @@ Component class testing should be kept very clean and simple.
 At first glance, you should be able to understand what the test is testing.
 
-Consider this `LightswitchComponent` which toggles a light on and off (represented by an on-screen message) when the user clicks the button.
+Consider this `LightswitchComponent` which toggles a light on and off \(represented by an on-screen message\) when the user clicks the button.
```
```suggestion
Consider this `LightswitchComponent`, which toggles a light on and off \(represented by an on-screen message\) when the user clicks the button.
```

josmar-crwdstffng replied:
Updated.

```markdown
Consider this `LightswitchComponent`, which toggles a light on and off \(represented by an on-screen message\) when the user clicks the button.
```

TMDavisGoogle commented on aio/content/guide/testing-components-basics.md:
```diff
@@ -90,15 +90,15 @@ But many components have complex interactions with the DOM elements described in
 To answer these kinds of questions, you have to create the DOM elements associated with the components, you must examine the DOM to confirm that component state displays properly at the appropriate times, and you must simulate user interaction with the screen to determine whether those interactions cause the component to behave as expected.
 
-To write these kinds of test, you'll use additional features of the `TestBed` as well as other testing helpers.
+To write these kinds of test, you will use additional features of the `TestBed` as well as other testing helpers.
```
```suggestion
To write these kinds of tests, you will use additional features of the `TestBed` as well as other testing helpers.
```

TMDavisGoogle commented on aio/content/guide/testing-components-basics.md:
```diff
@@ -136,7 +136,7 @@ In this example, the metadata object passed to `TestBed.configureTestingModule`
 The default test module is pre-configured with something like the `BrowserModule` from `@angular/platform-browser`.
 
-Later you'll call `TestBed.configureTestingModule()` with imports, providers, and more declarations to suit your testing needs.
+Later you will call `TestBed.configureTestingModule()` with imports, providers, and more declarations to suit your testing needs.
```
```suggestion
Later, you will call `TestBed.configureTestingModule()` with imports, providers, and more declarations to suit your testing needs.
```

josmar-crwdstffng replied:
Updated.

```markdown
Later, use `TestBed.configureTestingModule()` with imports, providers, and more declarations to suit your testing needs.
```

TMDavisGoogle commented on aio/content/guide/testing-components-basics.md:
```diff
@@ -188,7 +188,7 @@ Now add a test that gets the component's element from `fixture.nativeElement` an
 
 The value of `ComponentFixture.nativeElement` has the `any` type.
-Later you'll encounter the `DebugElement.nativeElement` and it too has the `any` type.
+Later you will encounter the `DebugElement.nativeElement` and it too has the `any` type.
```
```suggestion
Later, you will encounter the `DebugElement.nativeElement`, and it also has the `any` type.
```

josmar-crwdstffng replied:
Updated.

```markdown
Later, you encounter the `DebugElement.nativeElement`, and it also has the `any` type.
```

TMDavisGoogle commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -357,14 +357,15 @@ Here are the code files discussed on this page.
-
-@reviewed 2022-02-28
+* You refactored data access to the `HeroService` class.
+* You registered the `HeroService` as the *provider* of its service at the root level so that it can be injected anywhere in the application.
```
```suggestion
* You registered the `HeroService` as the *provider* of its service at the root level, so that it can be injected anywhere in the application.
```

josmar-crwdstffng replied:
Updated.

```markdown
*   You registered the `HeroService` as the *provider* of its service at the root level, so that it can be injected anywhere in the application
```

TMDavisGoogle reviewed (changes requested):
Second pass: light edits (punctuation, grammar) for better clarity.

josmar-crwdstffng commented on aio/content/guide/component-styles-old.md:
```diff
@@ -0,0 +1,287 @@
+
+Sometimes it's useful to apply styles to elements within a component's template 
+based on some condition in an element that is an ancestor of the host element.
+For example, a CSS theme class could be applied to the document &grave;&lt;body&gt;&grave; element, and
```
This topic is being removed.

dylhunn commented:
I am concerned about the size of this PR. Due to the extraordinarily large size, it's effectively unreviewable.

Was this PR automatically generated? It seems to affect a lot of files we don't want to touch, such as `CHANGELOG.md`.

jelbourn commented on aio/content/guide/architecture-next-steps.md:
```diff
@@ -3,71 +3,143 @@
 
 <!-- links -->
 
+[AioCli]: cli "CLI Overview and Command Reference | Angular"
```
I'm concerned these titles will cause extra noise for screen-reader users.  For example, on this link (based on how it's used) a screen-reader would announce:
```
CLI Command Reference CLI Overview and Command Reference Angular
```
Which is more confusing than if it were just the link text of "CLI Command Reference"

jelbourn commented on aio/content/guide/architecture.md:
```diff
@@ -162,8 +162,7 @@ The following diagram shows how these basic pieces are related.
 *   [Introduction to Modules](guide/architecture-modules)
-*   [Introduction to Components](guide/architecture-components)
-    *   [Templates and views](guide/architecture-components#templates-and-views)
+*   [Introduction to Components](guide/architecture-components)<!--*   [Templates and views](guide/architecture-components#templates-and-views)-->
```
What's this comment?

josmar-crwdstffng replied:
@jelbourn 
The content was commented out to retain location of previous information during peer and technical review.
All non-todo comments should be removed prior to publication.

jelbourn commented on aio/content/guide/component-interaction-old.md:
```diff
@@ -0,0 +1,363 @@
+# Component interaction
```
Should this file be deleted?

josmar-crwdstffng replied:
Yes, this topic is retained for reference only during the review.

jelbourn commented on aio/content/guide/component-overview-old.md:
```diff
@@ -0,0 +1,189 @@
+
+See [Angular component overview](guide/component/component-overview "Angular component overview | Angular").
+
+## old content
```
Should this file be deleted?

(same for other files marked as -"old")

josmar-crwdstffng replied:
Yes, this topic is retained for reference only during the review.

jelbourn commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+
+## Learn about Angular components
+
+<div class="card-container">
```
I don't think this card-styled set of links looks particularly good in a guide page like this vs. a higher-level landing page.

jelbourn reviewed (commented):
+1 to @dylhunn - this PR is effectively unreviewable. 

I've tried to read through some of it, and have left some comments based on what I could see, but it's impossible to tell which content was already here and moved versus new content being written for this PR.

josmar-crwdstffng commented on aio/content/guide/architecture-components.md:
```diff
@@ -1,95 +1,101 @@
-For example, individual components define and control each of the following views from the [Tour of Heroes tutorial](tutorial):
+<div  class="alert is-helpful">
+
+<header><b>Example</b>: Individual components define and control each of the following views from the <a href="tutorial" alt="Tour of Heroes app and tutorial | Angular">Tour of Heroes tutorial</a></header>
```
Html tags can be inserted in markdown without issue.
When you use an html tag, you should continue to use html until the html tag is closed in order to eliminate conflicts between the parsing of markdown and passing through of html style definitions.

dylhunn commented:
> it's impossible to tell which content was already here and moved versus new content being written for this PR

I agree with this as well -- I think we should put moves and additions in separate PRs.

josmar-crwdstffng committed:
docs: move component content topics

Deleted old component content files.
Added temporary stub files for old component content files.
Added new component content files.
Updated navigation file for new component content files.
Updated pullapprove file for new and updated component content files.
Updated firebase file for new and updated redirects.
Commented-out redirects in firebase file that require support for anchor-level links.

josmar-crwdstffng committed:
docs: remove unused docregions

Remove unused docregion for each code-example.

josmar-crwdstffng committed:
docs: update links per component content changes

Updated links in markdown files to align with changes to component content topics.
Updated links in readme files to align with changes to component content topics.
Updated links in typescript files to align with changes to component content topics.
Updated links in elements content to align with changes to component content topics.
Updated links in errors content to align with changes to component content topics.
Updated links in start content to align with changes to component content topics.
Updated links in tests content to align with changes to component content topics.
Updated links in tutorial content to align with changes to component content topics.

josmar-crwdstffng committed:
docs: removed old left-nav

Removed links to old component content topics from left-nav TOC.

mary-poppins commented:
You can preview 6a7c174 at https://pr45501-6a7c174.ngbuilds.io/.

mary-poppins commented:
You can preview 9d42558 at https://pr45501-9d42558.ngbuilds.io/.

josmar-crwdstffng commented on aio/content/errors/NG0100.md:
```diff
@@ -1,30 +1,35 @@
 @name Expression Changed After Checked
+
```
@jelbourn 
The updates have been moved into another branch.

josmar-crwdstffng commented on aio/content/guide/architecture-components.md:
```diff
@@ -1,95 +1,101 @@
 # Introduction to components and templates
+<!-- structure but point to component structure for details -->
```
@jelbourn 
The comment was updated to reflect changes in the content structure.

josmar-crwdstffng commented on aio/content/cli/index.md:
```diff
@@ -83,11 +105,17 @@ Option names in the configuration file must use [camelCase](guide/glossary#case-
+
+ng {commandNameOrAlias} {requiredArguments} [{optionalArguments}] [{options}]
+   &bsol;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;/ &bsol;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;/ &bsol;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowbar;&lowba ...
+    &verbar;                    &verbar;                   &verbar;                     &verbar;
```
@jelbourn 
The updates have been moved into another branch.

josmar-crwdstffng commented on aio/content/guide/architecture-components.md:
```diff
@@ -1,95 +1,101 @@
 <div class="lightbox">
 
-<img alt="Metadata" class="left" src="generated/images/guide/architecture/metadata.png">
+<img alt="Metadata" class="left" src="generated/images/guide/architecture/metadata.png" />
```
@jelbourn 
I reverted most of the changes.
The changes were suggested to reflect closure activity found in the `angular/angular` repo.

TMDavisGoogle commented on aio/content/errors/NG0100.md:
```diff
@@ -15,16 +15,18 @@ This error commonly occurs when you've added template expressions or have begun
 The [source maps](https://developer.mozilla.org/docs/Tools/Debugger/How_to/Use_a_source_map) generated by the CLI are very useful when debugging. Navigate up the call stack until you find a template expression where the value displayed in the error has changed.
 
-Ensure that there are no changes to the bindings in the template after change detection is run. This often means refactoring to use the correct [component lifecycle hook](guide/lifecycle-hooks) for your use case. If the issue exists within `ngAfterViewInit`, the recommended solution is to use a constructor or `ngOnInit` to set initial values, or use `ngAfterContentInit` for other value bindings.
+Ensure that there are no changes to the bindings in the template after change detection is run. This often means refactoring to use the correct [component lifecycle hook][AioGuideComponentUsageLifecycleHooks]for your use case. If the issue exists within `ngAfterViewInit`, the recommended solution is to use a constructor or `ngOnInit` to set initial values, or use `ngAfterContentInit` for other va ...
```
```suggestion
Ensure that there are no changes to the bindings in the template after change detection is run. This often means refactoring to use the correct [component lifecycle hook][AioGuideComponentUsageLifecycleHooks] for your use case. If the issue exists within `ngAfterViewInit`, the recommended solution is to use a constructor or `ngOnInit` to set initial values, or use `ngAfterContentInit` for other value bindings.
```

josmar-crwdstffng replied:
Updated.

```md
Ensure that there are no changes to the bindings in the template after change detection is run. This often means refactoring to use the correct [component lifecycle hook][AioGuideComponentUsageLifecycleHooks] for your use case. If the issue exists within `ngAfterViewInit`, the recommended solution is to use a constructor or `ngOnInit` to set initial values, or use `ngAfterContentInit` for other value bindings.
```

TMDavisGoogle commented on aio/content/guide/component/component-encapsulate-style.md:
```diff
@@ -0,0 +1,177 @@
+</code-example>
+
+These styles are post-processed so that each CSS selector is augmented with the appropriate `_nghost` or `_ngcontent` attribute.
+These modified selectors make sure the styles to be applied to the views of the components in an isolated and targeted fashion.
```
```suggestion
These modified selectors make sure that the styles are to be applied to the views of the components in an isolated and targeted fashion.
```

josmar-crwdstffng replied:
Updated.

```md
These modified selectors ensure that the styles to be applied to the views of the components are in an isolated and targeted fashion.
```

TMDavisGoogle reviewed (commented):
LGTM.

josmar-crwdstffng committed:
docs: update content per edit review feedback

Updated whitespace and phrasing per edit review feedback.

TMDavisGoogle reviewed (commented):
LGTM.

mary-poppins commented:
You can preview a4dfd63 at https://pr45501-a4dfd63.ngbuilds.io/.

mary-poppins commented:
You can preview 895cc77 at https://pr45501-895cc77.ngbuilds.io/.

jelbourn reviewed (commented):
I've started reviewing, but this PR is still massive with no clear way to tell what is new content and what is moved existing content. From a cursory scan there does appear to be a fair amount of new content, so I'm comparing side-by-side with the old content. I will continue when I can, but at this pace it may be well over a week until I get can through it all.

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+A [component][AioGuideGlossaryComponent] is the main building block for an Angular application.
+Every Angular application has at least one component, the *root component*, that connects a component hierarchy with the document object model \(DOM\) of the page.
+
+It is the class that controls the portion of a browser screen identified as a [view][AioGuideGlossaryView].
```
Updated.

```md
The root component is the class that controls the portion of a browser screen identified as a [view][AioGuideGlossaryView].
```

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+A [component][AioGuideGlossaryComponent] is the main building block for an Angular application.
+Every Angular application has at least one component, the *root component*, that connects a component hierarchy with the document object model \(DOM\) of the page.
+
+It is the class that controls the portion of a browser screen identified as a [view][AioGuideGlossaryView].
```
Reworked and moved.

```md
A component contains the data and logic metadata for your Angular application
A component contains the metadata for an associated HTML [template][AioGuideTemplateOverview].

Angular uses a component and the associated HTML [template][AioGuideTemplateOverview] to render a collection of DOM elements.
The collection of DOM elements define an Angular [view][AioGuideGlossaryView].
An Angular [view][AioGuideGlossaryView] is the portion of a browser screen.
A component and the associated HTML template define and control an Angular [view][AioGuideGlossaryView].
```

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+Each component defines a class that contains application data and logic.
+Each component is associated with an HTML *template* that defines and controls the portion of a browser screen identified as a [view][AioGuideGlossaryView].
+
+The `@Component()` decorator identifies the class immediately below it as a component, and provides the template and related component-specific metadata.
```
Updated.

```md
A component is a TypeScript class that is defined using the the `@Component()` decorator.
```

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+
+<div class="alert is-helpful">
+
+Decorators are functions that modify JavaScript classes. Angular defines a number of decorators that attach specific kinds of metadata to classes, so that the system knows what those classes mean and how they should work.
```
Updated.

```md
<div class="alert is-helpful">

The `@Component` decorator is a [decorator][TypescriptlangDocsHandbookDecoratorsHtmlDecorators] that augments your TypeScript class with additional Angular-specific information.
The Angular-specific information defines how the Angular framework renders the component in the document object model \(DOM\).

</div>
```

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,84 @@
+
+To learn more about decorators on the web, see [Exploring EcmaScript Decorators](https://medium.com/google-developers/exploring-es7-decorators-76ecb65fb841#.x5c2ndtx0)
+
+[MediumGoogleDevelopersExploring-es7-decorators-76ecb65fb841#.x5c2ndtx0]: https://medium.com/google-developers/exploring-es7-decorators-76ecb65fb841#.x5c2ndtx0 "Exploring EcmaScript Decorators | medium.com"
```
Removed.

josmar-crwdstffng committed:
docs: update title for component-overview

Updated title for component/component-overview file.

josmar-crwdstffng committed:
docs: update content per technical review feedback

Updated phrasing and links per technical review feedback.

mary-poppins commented:
You can preview 23f5c67 at https://pr45501-23f5c67.ngbuilds.io/.

josmar-crwdstffng committed:
docs: modify reference link

Modified reference links to temporarily comment out alternate text.

mary-poppins commented:
You can preview acc7dcb at https://pr45501-acc7dcb.ngbuilds.io/.

mary-poppins commented:
You can preview 6dfb376 at https://pr45501-6dfb376.ngbuilds.io/.

jelbourn commented on aio/content/navigation.json:
```diff
@@ -108,53 +108,126 @@
+              ]
+            },
+            {
+              "title": "Component projection",
```
I would prefer to keep the content projection topic all in one page. Looking at the production content on aio today, it seems like it wouldn't be too much work to tweak the content to not be tutorial-ified / step-by-step; the vast majority of the content is overview/guide style, it just has a few paragraphs that devolve into step territory.

josmar-crwdstffng replied:
@jelbourn

Tearing apart the tutorials to create anonymized task-based topics while retaining the overview topic.

josmar-crwdstffng committed:
docs: remove temp files

Removed temporary `-old.md` files.

josmar-crwdstffng commented on aio/content/guide/component-overview-old.md:
```diff
@@ -0,0 +1,5 @@
+
+See [Angular component overview](guide/component/component-overview "Understand Angular components | Angular").
+
+@reviewed 2022-05-17
```
@jelbourn 

Removed file.

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,94 @@
+<div class="alert is-helpful">
+
+The `@Component` decorator is a [decorator][TypescriptlangDocsHandbookDecoratorsHtmlDecorators] that augments your TypeScript class with additional Angular-specific information.
+The Angular-specific information defines how the Angular framework renders the component in the document object model \(DOM\).
```
@jelbourn 

Updated.

```md
The `@Component` statement is a [decorator][TypescriptlangDocsHandbookDecoratorsHtmlDecorators] that augments your TypeScript class with Angular-specific information.
The information defines how the Angular framework renders the component in the document object model \(DOM\).
```

josmar-crwdstffng commented on aio/content/guide/component/component-overview.md:
```diff
@@ -0,0 +1,94 @@
+An Angular [view][AioGuideGlossaryView] is the portion of a browser screen.
+A component and the associated HTML template define and control an Angular [view][AioGuideGlossaryView].
+
+When a user interacts with your application, Angular uses APIs to create, update, and remove [components][AioGuideGlossaryComponent].
```
@jelbourn 

Updated.

```md
The Angular framework uses a [component][AioGuideGlossaryComponent] and the associated HTML [template][AioGuideTemplateOverview] to complete the following actions.

*   Render the HTML template in the DOM structure
*   Update the rendered DOM structure
*   Remove a component and the associated DOM structure

The Angular framework updates your component when a user interacts with your application.
```

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+# Understand the structure an Angular component
```
@jelbourn 

I reworked the page to include your suggestions and describe all parts of a component.

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+# Understand the structure an Angular component
+
+The logic of a [component][AioGuideGlossaryComponent] class uses methods and properties in an API to define and interact with the [view][AioGuideGlossaryView].
```
@jelbourn 

I replaced the `view` term with the `rendered DOM elements` or `HTML template` as appropriate.

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+| Template HTML       | An HTML file or inline HTML template | Declares what is rendered on a webpage. <br /> To learn more, see [Template syntax][AioGuideTemplateSyntax]. |
+| Component class     | A Typescript file                    | Defines behavior.                                                                                            |
+| Inline CSS selector | In the template HTML                 | Defines how the [component][AioGuideGlossaryComponent] is used in a [template][AioGuideGlossaryTemplate].    |
+| Inline CSS          | In the template HTML                 | **Optional**. <br /> The CSS styles directly applied to the template.                                        |
```
@jelbourn

I restructured the component and metadata tables to better scope the parts.

Updated.

```md
Each [component][AioGuideGlossaryComponent] class must have the following parts.

| Part                   | Details |
|:---                    |:---     |
| `@component` decorator | Informs the Angular framework that Angular-specific information is included in the TypeScript class. |
| metadata object        | Includes the `selector` and other properties for the component class.                                |
```

```md
Each component in your application has separate metadata and allows you to use a different combination of inline and file references for each component.

The metadata for each component must include the `selector` property.

The following table lists the properties of the metadata object.

| Property                               | Details |
|:---                                    |:---     |
| [`selector`](#selector-property)       | **REQUIRED**. The name of the CSS selector the Angular framework replaces in the HTML [template][AioGuideGlossaryTemplate] with an instance of the component. |
| [`template`](#template-property)       | The inline HTML template of the component. If you specify the `templateUrl` property, you must not specify the `template` property.                                 |
| [`templateUrl`](#templateurl-property) | The relative path to the HTML template file of the component. If you specify the `template` property, you must not specify the `templateUrl` property.              |
| [`styles`](#styles-property)           | The inline CSS styles for the template of the component.                                                                                                      |
| [`styleUrls`](#styleurls-property)     | The relative path to each CSS style file for the template of the component.                                                                                   |
| [`providers`](#providers-property)     | An array of [providers][AioGuideGlossaryProvider] for services that the component requires.                                                                   |
```

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+You are able to interact with the lifecycle of a component at each moment [using the optional lifecycle hook methods][AioGuideComponentLifecycleUseLifecycleHookMethod].
+For example, the `ngOnInit()` method.
+
+</div>
```
@jelbourn 

Updated.

```md
<div class="alert is-helpful">

**TIP**: <br />
The Angular framework provides [lifecycle hook methods][AioGuideComponentLifecycleUseLifecycleHookMethod] to access [different phases of the rendering process][AioGuideComponentLifecycle] for each component.
To learn more about the different phases of the rendering process, see [Understand the lifecycle of a component][AioGuideComponentLifecycle].
To learn more about the lifecycle hook methods, see [Use lifecycle hook method][AioGuideComponentLifecycleUseLifecycleHookMethod].

</div>
```

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+
+## Component class
+
+A [component][AioGuideGlossaryComponent] is a specific type of [directive][AioGuideGlossaryDirective].
```
@jelbourn 

I removed the discussion of the relationship between a directive and component.

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+
+</div>
+
+The component metadata associates the HTML [template][AioGuideGlossaryTemplate] with the `@Component()`.
```
@jelbourn

I restructured the component and metadata tables to better scope the parts.

Updated.

```md
Each [component][AioGuideGlossaryComponent] class must have the following parts.

| Part                   | Details |
|:---                    |:---     |
| `@component` decorator | Informs the Angular framework that Angular-specific information is included in the TypeScript class. |
| metadata object        | Includes the `selector` and other properties for the component class.                                |
```

```md
Each component in your application has separate metadata and allows you to use a different combination of inline and file references for each component.

The metadata for each component must include the `selector` property.

The following table lists the properties of the metadata object.

| Property                               | Details |
|:---                                    |:---     |
| [`selector`](#selector-property)       | **REQUIRED**. The name of the CSS selector the Angular framework replaces in the HTML [template][AioGuideGlossaryTemplate] with an instance of the component. |
| [`template`](#template-property)       | The inline HTML template of the component. If you specify the `templateUrl` property, you must not specify the `template` property.                                 |
| [`templateUrl`](#templateurl-property) | The relative path to the HTML template file of the component. If you specify the `template` property, you must not specify the `templateUrl` property.              |
| [`styles`](#styles-property)           | The inline CSS styles for the template of the component.                                                                                                      |
| [`styleUrls`](#styleurls-property)     | The relative path to each CSS style file for the template of the component.                                                                                   |
| [`providers`](#providers-property)     | An array of [providers][AioGuideGlossaryProvider] for services that the component requires.                                                                   |
```

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+| [`providers`](#providers)           | An array of [providers][AioGuideGlossaryProvider] for services that the component requires.  |
+| [`selector`](#selector)             | An HTML tag element.                                                                         |
+| [`styles`](#style-metadata)         | The inline CSS styles for the template of the component.                                     |
+| [`styleUrls`](#style-metadata)      | The module-relative address of each CSS style file for the template of the component.        |
```
@jelbourn 

Replaced instances of the `module-relative address of` phrase with the `relative path to` phrase.

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+
+<div class="lightbox">
+
+<img alt="Metadata" class="left" src="generated/images/guide/architecture/metadata.png" />
```
@jelbourn 

Yes.
Removed.

josmar-crwdstffng commented on aio/content/guide/component/component-structure.md:
```diff
@@ -0,0 +1,189 @@
+
+<code-example format="typescript" header="Class with no special Angular notation or syntax" language="typescript">
+
+export class &lcub;NameOfComponent&rcub;Component implements OnInit {
```
Task-based topics should be be constructed to address the functionality with descriptive names.
The use of a specific scenario fits a procedure or tutorial topic, but is not useful in a task-based topic.

josmar-crwdstffng commented on aio/content/guide/component/component-lifecycle.md:
```diff
@@ -0,0 +1,39 @@
+# Understand the lifecycle of a component
```
@jelbourn 

Removed redundant item form the left-nav TOC.

josmar-crwdstffng commented on aio/content/navigation.json:
```diff
@@ -108,53 +108,126 @@
+               ]
+            },
+            {
+              "title": "Component data",
```
@jelbourn 

The left-nav is adaptable.
I placed the content where it appeared to fit in context.

Updated.

josmar-crwdstffng commented on aio/content/navigation.json:
```diff
@@ -108,53 +108,126 @@
         {
-          "title": "Components",
-          "tooltip": "Learn more about how to use data binding to build dynamic views",
+          "title": "Component",
```
One of the main tenets for the document restructure project is separating out the various content types into topics that focus on 1 content type.
This increases readability, reduces cognitive load, reduces maintenance, and increases consistency.

josmar-crwdstffng commented on aio/content/guide/component/component-web.md:
```diff
@@ -0,0 +1,274 @@
+# Web component
```
@jelbourn 

Moved the content under from `guide/component/component-web` to `guide/element/custom-element`.

josmar-crwdstffng commented on aio/content/guide/component/component-web.md:
```diff
@@ -0,0 +1,274 @@
+# Web component
```
@jelbourn 

Singular term is more clear.

The title was changed to `Understand custom element`.

josmar-crwdstffng commented on aio/content/guide/component/component-web.md:
```diff
@@ -0,0 +1,274 @@
+# Web component
+
+An Angular component packaged as a *custom element* is an *Angular element*.
+A *custom element* is also refernced as a [Web Component][MdnDocsWebWebComponents].
```
@jelbourn 

Updated.

```md
An *Angular element* is an Angular component packaged as a [custom element][MdnDocsWebWebComponentsUsingCustomElements].
A *custom element* is also called a [Web Component][MdnDocsWebWebComponents].
A web component is a custom element that also defines a [Shadow Root][MdnDocsWebApiShadowroot] interface.
[Web Components][MdnDocsWebWebComponents] are a web standard that allows you to define a new HTML element in a framework-agnostic way.
```

josmar-crwdstffng commented on aio/content/guide/component/component-example.md:
```diff
@@ -0,0 +1,252 @@
+# Example Angular component applications
```
@jelbourn 

Tearing apart the tutorials to create anonymized task-based topics.

josmar-crwdstffng commented on aio/content/guide/component/component-lifecycle.md:
```diff
@@ -0,0 +1,39 @@
+# Understand the lifecycle of a component
```
@jelbourn

Tearing apart the tutorials to create anonymized task-based topics.

mary-poppins commented:
You can preview d4996d5 at https://pr45501-d4996d5.ngbuilds.io/.

josmar-crwdstffng commented:
Closed PR to implement with feedback and comments.
